unity-client/Assets/Scripts/MainScripts/DCL/DataStore/Variables/BaseCollection.cs

```diff
@@ -0,0 +1,62 @@
+using System;
+using System.Collections.Generic;
+
+public class BaseCollection<T> : IBaseCollection<T>, IEquatable<IEnumerable<T>>
+{
+    public event Action<IEnumerable<T>> OnChange;
```

Comment:
I feel `OnChange` naming a little misleading here cause removing and/or adding doesn't trigger this event even if in some way the collection has changed. I can't come up with any suggestion right now tho

Reply:
What do you think of `OnSet`?